### PR TITLE
[WIP] Enable handlers

### PR DESCRIPTION
**Pull request by johrstrom**
Here's a bunch of code I'd been working on that I just got around to finishing.  This is supposed to be the baseline changes for #14 so that nodes can pass context of what happened back to the Schedule.  The actual Context struct is likely to be extended, but this may due for now.

### CODE REVIEW

**Comment by johrstrom** [comment]
Hey @kpolnitz I don't know if just requesting your approval got a notification, if not here's one.

If you're able, take you're time, I'm in no rush. Feel free to ask questions or even question what I've done because it may not be the optimal solution.

**Comment by kpolnitz** [comment]
Sorry, busy finals season just saw this! I especially like the addition of the locks.

File: config/config.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package config
 
 import (
+	"sync"
 	"time"
 
 	"github.com/sirupsen/logrus"
```
Comment (kpolnitz) [inline config/config.go:7]: Just curious: does "sync" also have support for spinlocks, etc. ?

**Review by kpolnitz — APPROVED** [review]

File: config/config.go (hunk 1 above)
Comment (johrstrom) [reply config/config.go:7]: Looks like they do their own thing, which kinda looks like spinlock but with a bit of control on the 'spinning'. 
https://golang.org/src/sync/mutex.go

but then they have this bit at the end, if(race.Enabled), which ends up pointing to the 'internal' package which is a bunch of platform specific stuff. (like a BSD specific type of mutex might actually be instantiated when you're on a BSD platform). 

here's just other docs i ended up reading this morning. 
https://kernel.readthedocs.io/en/sphinx-samples/kernel-locking.html

And thanks for looking into this and merging. There are tickets here if you ever get bored enough.

**Pull request merged**